ddl/ddl_api.go

```diff
@@ -3193,3 +3225,103 @@ func buildPartitionInfo(meta *model.TableInfo, d *ddl, spec *ast.AlterTableSpec)
 	}
 	return part, nil
 }
+
+// LockTables uses to execute lock tables statement.
+func (d *ddl) LockTables(ctx sessionctx.Context, stmt *ast.LockTablesStmt) error {
+	lockTables := make([]model.TableLockTpInfo, 0, len(stmt.TableLocks))
+	is := d.infoHandle.Get()
+	sessionInfo := model.SessionInfo{
+		ServerID:  d.GetID(),
+		SessionID: ctx.GetSessionVars().ConnectionID,
+	}
+	uniqueTableID := make(map[int64]struct{})
+	// Check whether the table was already locked by other.
+	for _, tl := range stmt.TableLocks {
+		tb := tl.Table
+		if tb.Schema.L == infoschema.LowerName || tb.Schema.L == "performance_schema" || tb.Schema.L == mysql.SystemDB {
```

Comment:
```suggestion
		if tb.Schema.L == infoschema.LowerName || tb.Schema.L == performance_schema.LowerName || tb.Schema.L == mysql.SystemDB {
```

Reply:
You will got "import cycle not allowed" error.

Reply:
All right, then... maybe we should fix it later.

Reply:
Add a TODO here?

Reply:
TODO what?

Reply:
replace "performance_schema" with `xxx.LowerName`